contracts/staking/BaseShareField.sol

```diff
@@ -37,8 +37,8 @@ contract BaseShareField is DAOContract{
         uint amount;     // How many tokens the user has provided.
         uint rewardDebt; // Reward debt. 
         uint rewardEarn; // Reward earn and not minted
```

Comment:
this can also be reduced
as theoretically there shuoldnt be more than 2.2 trillion G$
thats 1e12 + 00 (for decimals)

Reply:
Yes true but as we discussed I keep this values in higher decimals

Reply:
hmm maybe uint112 then?